### PR TITLE
54 speed up distance to road

### LINKED ISSUE #54: Speed-up distance_to_road
It's slow: https://github.com/nptscot/uptake/issues/12

### PR DESCRIPTION
**Pull request by Robinlovelace**
- **Remove union, fix #54**
- **Add test results for Edinburgh**

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -165,7 +167,10 @@ edinburgh_3km = edinburgh |>
 osm = get_travel_network("Scotland", boundary = edinburgh_3km, boundary_type = "clipsrc")
 cycle_net = get_cycling_network(osm)
 drive_net = get_driving_network_major(osm)
-cycle_net = distance_to_road(cycle_net, drive_net)
+system.time({
+  cycle_net = distance_to_road(cycle_net, drive_net)
```
Comment (Robinlovelace) [inline README.Rmd:171]: Heads-up @Hussein-Mahfouz I get the following error here on this branch:

```
Error: Incompatible lengths: 5803, 1021
```

I think we need the union somewhere, maybe we create a spatial grid and do it that way?

**Review by Robinlovelace — COMMENTED** [review]
Work in progress..

**Comment by Hussein-Mahfouz** [comment]
Glad to see this new version. Much faster than the initial one I proposed

**Comment by Robinlovelace** [comment]
Thanks Hussein!

**Pull request merged**